docsource/ops.rst

```diff
@@ -1,6 +1,228 @@
+
+.. note::
+
+    The commands below can be copy-pasted into Kibana and executed against the index. The corresponding curl commands can be obtained from Kibana by clicking on the little wrench icon that appears next to the pasted command.
+
+.. note::
+
+    If any of the steps below fail, don't proceed until the issue is resolved by referring to the ElasticSearch documentation.
+
+1. Note current time
+
+2. Stop any workers that are performing trickle updates on the index that needs to be reindexed.
+
+3. Refresh the index that needs reindexing to make sure it has been brought to a consistent state.
+
+.. code-block:: bash
+
+    POST /kbase.genome_1/_refresh
+
+4. Get a checksum for the index and record it in a separate file for later verification.
+
+.. code-block:: bash
+
+    GET /kbase.genome_1/_stats/docs,store
+
+5a. If mapping needs to be changed (for cases b, c, d above), get current index mapping.
+
+.. code-block:: bash
+
+    GET kbase.genome_1/_mapping
+
+5b. Copy-paste the mapping from the current index into the body section of the PUT command below and make the necessary field change (preferably one change per complete reindexing operation).
+
+It is a good practice to make the mapping strict ("dynamic": "strict") for each type (data and access) in the index. Strict mappings prevent the mapping from being modified dynamically during ingest time.
+
+Update the settings section below the mapping. The number of shards and replicas must be decided on based on your capacity planning rules. It is costly to change the number of shards, so choose wisely! Make sure not to exceed 600 shards for any node in the system. Increase number of replicas to improve availability.
+
+.. code-block:: bash
+
+    PUT kbase.genome_1_r1
+    {
+      "mappings": {
+        "data": {
+          "dynamic": "strict",
+          "_parent": {
+            "type": "access"
+          },
+          "_routing": {
+            "required": true
+          },
+          "properties": {
+            "accgrp": {
+              "type": "integer"
+            },
+            . . .
+          }
+        },
+        "access": {
+          "dynamic": "strict",
+          "properties": {
+            "extpub": {
+            "type": "integer"
+            },
+            . . .
+          }
+        }
+      },
+      "settings": {
+        "index": {
+          "number_of_shards": "5",
+          "number_of_replicas": "1"
+        }
+      }
+    }
+
+5c. If the mapping does not require any change but the document's meta data needs to be changed, use the `Painless <https://www.elastic.co/guide/en/elasticsearch/reference/5.4/modules-scripting-painless-syntax.html>`_ script to modify metadata. Setting version_type to external will cause Elasticsearch to preserve the version from the source, create any documents that are missing, and update any documents that have an older version in the destination index than they do in the source index.
```

Comment:
Define meta data in this context, not clear.

Reply:
Added in brackets "field values".

Reply:
Are you talking about the `_X` fields or the `key.X` fields? Or both?

Reply:
Ah, good point. Clarified as follows: 

 If the mapping does not require any change but the documents' field values (not including meta-data fields whose key names start with underscore) need to be changed, use the `Painless <https://www.elastic.co/guide/en/elasticsearch/reference/5.4/modules-scripting-painless-syntax.html>`_ script to modify metadata.